rastervision/backend/torch_utils/chip_classification/data.py

```diff
@@ -3,19 +3,49 @@
 
 import numpy as np
 import torch
+
 from torch.utils.data import DataLoader, Dataset
+from torch.utils.data.sampler import WeightedRandomSampler
 from albumentations.core.composition import Compose
 from albumentations.augmentations.transforms import (
     Blur, RandomRotate90, HorizontalFlip, VerticalFlip, GaussianBlur,
     GaussNoise, RGBShift, ToGray)
 
 from rastervision.backend.torch_utils.data import DataBunch
-from rastervision.backend.torch_utils.chip_classification.folder import (
-    ImageFolder)
+from rastervision.backend.torch_utils.chip_classification.folder import ImageFolder
 
 log = logging.getLogger(__name__)
 
 
+def calculate_oversampling_weights(imageFolder, rare_classes, desired_prob):
+    '''
+    Calculates weights tensor for oversampling
+
+    args:
+        imageFolder: instance of
```

Comment:
I would refactor this to take in the array of targets (ie. class ids) for the dataset, rather than the dataset itself. This way we could use it with any type of dataset.

Reply:
Done

Reply:
Now set it to take in a list of class labels/targets. That should work for the chip classification. But in case of semseg or objdet i'm not sure that we can use the function in this form. Should the function only be usable for chipclassification?